src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,80 @@
+                .flatMap(s -> Arrays.stream(s.split(",")))
+                .map(Integer::valueOf)
+                .filter(i -> i % 2 == 0)
+                .mapToInt(i -> i)
+                .min();
+        return n.orElseThrow(() -> new RuntimeException("Can't get min value from list: "
+                + "< Here is our input 'numbers' >"));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        for (int i = 0; i < numbers.size(); i++) {
+            if (i % 2 != 0) {
+                numbers.set(i, numbers.get(i) - 1);
+            }
+        }
+        OptionalDouble optionalDouble = numbers.stream()
+                .filter(n -> n % 2 != 0)
+                .mapToInt(l -> l)
+                .average();
+        return optionalDouble.orElseThrow(NoSuchElementException::new);
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        Predicate<Person> personPredicate = p -> p.getSex().equals(Person.Sex.MAN)
+                && p.getAge() >= fromAge
+                && p.getAge() <= toAge;
+
+        return peopleList.stream()
+                .filter(personPredicate)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> personPredicate = p -> (p.getSex().equals(Person.Sex.MAN)
+                && p.getAge() >= fromAge
+                && p.getAge() <= maleToAge)
+                || (p.getSex().equals(Person.Sex.WOMAN)
+                && p.getAge() >= fromAge
+                && p.getAge() <= femaleToAge);
```

Comment:
move to private method
```suggestion
        Predicate<Person> personPredicate = person -> (p.getSex().equals(Person.Sex.MAN)
                && p.getAge() >= fromAge
                && p.getAge() <= maleToAge)
                || (p.getSex().equals(Person.Sex.WOMAN)
                && p.getAge() >= fromAge
                && p.getAge() <= femaleToAge);
```